firefox_ui_harness/testcase.py

```diff
@@ -24,15 +24,23 @@ def setUp(self, *args, **kwargs):
         self.browser = self.windows.current
 
     def tearDown(self, *args, **kwargs):
-        self.prefs.restore_all_prefs()
-        MarionetteTestCase.tearDown(self, *args, **kwargs)
-
-        # This assertion should be run after all other tearDown code
-        # so that in case of a failure, further tests will not run
-        # in a state that is more inconsistent than necessary.
-        win_count = len(self.marionette.window_handles)
-        self.assertEqual(win_count, self._start_handle_count,
-                         "A test must not leak window handles. "
-                         "This test started the browser with %s open "
-                         "top level browsing contexts, but ended with %s." %
-                         (self._start_handle_count, win_count))
+        try:
+            # Marionette needs an existent window to be selected. Take the first
+            # browser window which usually should not have been closed
+            # TODO: We might have to make this more error prone in case the
+            # original window has been closed.
+            self.browser.switch_to()
```

Comment:
This case would work because we switch back to the chrome window. What will not work is if a test opens a new window, but closes the original browser window. It's clearly something to improve, but I would this push out until we have more knowledge in handling windows, tabs, and about puppeteer at all.